app/script_variant.py

```diff
@@ -20,7 +20,14 @@ def read_dictionary(file_name):
             with open(file_name, 'r') as f:
                 for line in f:
                     kanji, hiragana = line[:-1].split('\t')
-                    dictionary[kanji.decode('utf-8')] = hiragana.decode('utf-8')
+                    kanji = kanji.decode('utf-8')
+                    hiragana = hiragana.decode('utf-8')
+                    if kanji in dictionary:
+                        hiragana_list = dictionary[kanji]
+                        if not hiragana in hiragana_list:
```

Comment:
You can use "set" instead of an array. Then you don't need this if-statement. It removes duplicates automatically.